### ISSUE TITLE
Unable to release with crate meta description

### ISSUE THREAD

**Issue by j-mendez** [PyO3/maturin#2012]
### Bug Description

When deploying with Mac, the build has trouble merging the Cargo.toml description and fails to deploy.
The error:

```
📦 Built source distribution to /Users/me/Code/OSS/spider-py/target/wheels/spider_rs-0.0.32.tar.gz
🔐 Using credential in pypirc for upload
🚀 Uploading 2 packages
💥 maturin failed
  Caused by: 💥 Failed to upload "spider_rs-0.0.32-cp312-cp312-macosx_11_0_arm64.whl" (7.3 MB)
  Caused by: Failed to upload the wheel with status 400: <html>
 <head>
  <title>400 'summary' has invalid data. See https://packaging.python.org/specifications/core-metadata for more information.</title>
 </head>
 <body>
  <h1>400 'summary' has invalid data. See https://packaging.python.org/specifications/core-metadata for more information.</h1>
  The server could not comply with the request since it is either malformed or otherwise incorrect.<br/><br/>
&#x27;summary&#x27; has invalid data. See https://packaging.python.org/specifications/core-metadata for more information.


 </body>
</html>
```

### Your maturin version (`maturin --version`)

1.5.0

### Your Python version (`python -V`)

3.11.5

### Your pip version (`pip -V`)

 24.0

### What bindings you're using

pyo3 v0.20.3

### Does `cargo build` work?

- [ ] Yes, it works

The error on build:

```
was built for newer 'macOS' version (14.4) than being linked (14.0)
```

### If on windows, have you checked that you aren't accidentally using unix path (those with the forward slash `/`)?

- [ ] Yes

### Steps to Reproduce

1. Setup crate with Cargo.toml that includes a description ex:
```
[package]
edition = "2021"
name = "spider_rs"
version = "0.0.32"
repository = "https://github.com/spider-rs/spider-py"
license = "MIT"
description = "The fastest web crawler and indexer."

[lib]
crate-type = ["cdylib"]

[dependencies]
indexmap = "2.1.0"
num_cpus = "1.16.0"
spider = { version = "1.88.2", features = ["budget", "cron", "regex", "cookies", "socks", "chrome", "control", "smart", "chrome_intercept", "cache", "serde", "openai" ] }
pyo3 = { version = "0.20.3", features = ["extension-module", "serde"] }
pyo3-asyncio = { version = "0.20", features = ["attributes", "tokio-runtime"] }
serde_json = "1.0.114"

[target.x86_64-unknown-linux-gnu.dependencies]
openssl-sys = { version = "0.9.96", features = ["vendored"] }
openssl = { version = "0.10.60", features = ["vendored"] }

[target.x86_64-unknown-linux-musl.dependencies]
openssl-sys = { version = "0.9.96", features = ["vendored"] }
openssl = { version = "0.10.60", features = ["vendored"] }

[profile.release]
lto = true
```
2. `maturin publish`

**Comment by messense** [comment]
We won't be able to help you unless you provide detailed reproduce steps and error messages.

**Comment by j-mendez** [comment]
@messense got it, will send more details shortly.

**Comment by j-mendez** [comment]
> 📦 Built source distribution to /Users/me/Code/OSS/spider-py/target/wheels/spider_rs-0.0.32.tar.gz
> 🔐 Using credential in pypirc for upload
> 🚀 Uploading 2 packages
> 💥 maturin failed
>   Caused by: 💥 Failed to upload "spider_rs-0.0.32-cp312-cp312-macosx_11_0_arm64.whl" (7.3 MB)
>   Caused by: Failed to upload the wheel with status 400: <html>
>  <head>
>   <title>400 'summary' has invalid data. See https://packaging.python.org/specifications/core-metadata for more information.</title>
>  </head>
>  <body>
>   <h1>400 'summary' has invalid data. See https://packaging.python.org/specifications/core-metadata for more information.</h1>
>   The server could not comply with the request since it is either malformed or otherwise incorrect.<br/><br/>
> &#x27;summary&#x27; has invalid data. See https://packaging.python.org/specifications/core-metadata for more information.
> 
> 
>  </body>
> </html>

updated.

**Comment by messense** [comment]
Duplicate of #1998? Can you try maturin v1.5.1?

**Comment by j-mendez** [comment]
> maturin --version

fixed in `v1.5.1` thank you!
